### PR TITLE
feat(dashboard): Tag input suggestions

### PR DESCRIPTION
**Pull request by desiprisg**
### What changed? Why was the change needed?

* Removed barrel files
* Added the suggestions for tag inputs
### Screenshots

https://github.com/user-attachments/assets/aed7024e-cdaa-4af4-b5ec-2709d7dfd8fa

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/create-workflow-button.tsx [hunk 1]
```diff
@@ -54,6 +55,12 @@ export const CreateWorkflowButton = (props: CreateWorkflowButtonProps) => {
       setIsOpen(false);
     },
   });
+  const { data: tagData } = useQuery<{ data: { name: string }[] }>({
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/create-workflow-button.tsx]: Let's factor this out to a dedicated `useTags` hook. We will need it in other screens as well.

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 2]
```diff
@@ -39,47 +45,74 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
...
-              onClick={() => removeTag(tag)}
-              className="ml-1 rounded-full outline-none focus:outline-none"
-            >
-              <RiCloseFill className="size-3" />
-              <span className="sr-only">Remove tag</span>
-            </button>
-          </Badge>
-        ))}
-      </div>
+    <div>
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: This extra div can be removed

File: apps/dashboard/src/components/primitives/tag-input.tsx [hunk 3]
```diff
@@ -39,47 +45,74 @@ const TagInput = forwardRef<HTMLInputElement, TagInputProps>((props, ref) => {
...
+            portal={false}
+            onOpenAutoFocus={(e) => {
+              e.preventDefault();
+            }}
+          >
+            <CommandList>
+              <CommandGroup>
+                {inputValue !== '' && (
+                  <CommandItem
+                    value={`input-${inputValue}`} // if this exists in the list below there are bugs introduces to the key up/down behaviour
```
Comment (SokratisVidros) [inline apps/dashboard/src/components/primitives/tag-input.tsx]: @desiprisg This comment doesn't make a lot of sense. Can you make it more descriptive?

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/primitives/tag-input.tsx (hunk 2 above)
Comment (desiprisg) [reply apps/dashboard/src/components/primitives/tag-input.tsx]: Done

File: apps/dashboard/src/components/create-workflow-button.tsx (hunk 1 above)
Comment (desiprisg) [reply apps/dashboard/src/components/create-workflow-button.tsx]: Done. I added a `useTagsQuery` here and I think that pattern will suit us better than filtering out what we pass / return. @LetItRock  Let's discuss this and see what pattern will fit us best.

File: apps/dashboard/src/components/primitives/tag-input.tsx (hunk 3 above)
Comment (desiprisg) [reply apps/dashboard/src/components/primitives/tag-input.tsx]: Done

File: apps/dashboard/src/hooks/use-tags-query.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { useQuery } from '@tanstack/react-query';
+import { QueryKeys } from '@/utils/query-keys';
+import { useEnvironment } from '@/context/environment/hooks';
+import { getV2 } from '@/api/api.client';
+
+export const useTagsQuery = () => {
+  const { currentEnvironment } = useEnvironment();
+  const query = useQuery<{ data: { name: string }[] }>({
+    queryKey: [QueryKeys.fetchWorkflow, currentEnvironment?._id],
+    queryFn: async () => await getV2(`/environments/${currentEnvironment!._id}/tags`),
```
Comment (LetItRock) [inline apps/dashboard/src/hooks/use-tags-query.ts:10]: please move it to the API helper function, then we won't need to do `data.data` when using the query hook

**Review by LetItRock — APPROVED** [review]

File: apps/dashboard/src/components/create-workflow-button.tsx (hunk 1 above)
Comment (LetItRock) [reply apps/dashboard/src/components/create-workflow-button.tsx]: @desiprisg I'm totally fine with following `useXQuery, useXMutation` pattern ;) let's just quickly update the hooks in the separate ticket

Comment (SokratisVidros) [reply apps/dashboard/src/components/create-workflow-button.tsx]: Folks, the one we have now is OK as well. So please align them separately.

Comment (desiprisg) [reply apps/dashboard/src/components/create-workflow-button.tsx]: I also meant not passing `onSuccess` and `onError` funcs, and just passing through what `@tanstack/react-query` accepts.

File: apps/dashboard/src/hooks/use-tags-query.ts (hunk 4 above)
Comment (desiprisg) [reply apps/dashboard/src/hooks/use-tags-query.ts:10]: I don't think this is correct. If it's a `useTagsQuery` hook, then you should get back whatever was in the body, right? If `data` doesn't make sense, then that's a BE concern I guess.

Comment (desiprisg) [reply apps/dashboard/src/hooks/use-tags-query.ts:10]: merged in order to get the import changes in. If you feel strongly about this, let's discuss privately.

**Pull request merged**